herddb-core/src/main/java/herddb/file/FileDataStorageManager.java

```diff
@@ -236,14 +259,40 @@ private boolean isTableOrIndexCheckpointsFile(Path path) {
         return result;
     }
 
+    private static List<Record> rawReadDataPage(Path pageFile, InputStream stream) throws IOException, DataStorageManagerException {
+        List<Record> result;
+        long hashFromFile;
+        long hashFromDigest;
+        try (XXHash64Utils.HashingStream hash = new XXHash64Utils.HashingStream(stream);
+                ExtendedDataInputStream dataIn = new ExtendedDataInputStream(hash)) {
+            long version = dataIn.readVLong(); // version
+            long flags = dataIn.readVLong(); // flags for future implementations
+            if (version != 1 || flags != 0) {
+                throw new DataStorageManagerException("corrupted data file " + pageFile.toAbsolutePath());
+            }
+            int numRecords = dataIn.readInt();
+            result = new ArrayList<>(numRecords);
+            for (int i = 0; i < numRecords; i++) {
+                byte[] key = dataIn.readArray();
+                byte[] value = dataIn.readArray();
+                result.add(new Record(new Bytes(key), new Bytes(value)));
+            }
+            hashFromDigest = hash.hash();
+            hashFromFile = dataIn.readLong();
```

Comment:
Can we add a comment about padding?
The fille will be padded with zeroes

Reply:
sure we can add some comment... padding itself isn't a problem here because the reader already known how much data need to read and stops without reaching padding or eof (if the file isn't truncated obviously)

Reply:
Some time ago I was trying to change the read procedure in order to look for the checksum at the end of the file.
But it was not possible because we are already rewriting pages and we have already churn at the end of data pages

Reply:
Added a comment about while padding doesn't matters